androidApp/src/main/java/com/example/tasky/android/agenda/viewmodel/AgendaViewModel.kt

```diff
@@ -24,28 +23,15 @@ class AgendaViewModel(
     fun getAgendas(timeStamp: Long) {
         viewModelScope.launch {
             agendaRepository.getAgenda(timeStamp = timeStamp).onSuccess { data ->
-                val list = mutableListOf<AgendaItem>()
-                list.addAll(data.events)
-                list.addAll(data.tasks)
-                list.addAll(data.reminders)
-                list.sortWith(
-                    java.util.Comparator { item1, item2 ->
-                        val time1 =
-                            when (item1) {
-                                is Event -> item1.from
-                                is Task -> item1.time
-                                is Reminder -> item1.time
+                val list =
+                    (data.events + data.tasks + data.reminders)
+                        .sortedBy {
+                            when (it) {
+                                is Event -> it.from
+                                is Task -> it.time
+                                is Reminder -> it.time
```

Comment:
Given that all agenda items have some sort of start time, could it make sense sharing that property? Then you would not need this when expression, but could just sort by `it.startTime`